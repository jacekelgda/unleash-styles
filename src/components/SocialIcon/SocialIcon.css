.Icon {
  display: inline-block;
}
.Icon:not(:last-child) {
	margin-right: 20px;
}
.IconLinkPath {
	fill: #b2b6bc;
  fill-rule: evenodd;
  opacity: 0.5;
  transition: opacity .2s ease-out;
}
.IconLink:hover .IconLink {
	opacity: 1;
}
