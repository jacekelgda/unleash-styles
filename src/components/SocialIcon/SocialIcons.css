@media screen and (max-width: 767px) {
	.SocialIcons {
		display: none;
	}
}

.SocialIcons {
	display: inline-flex;
	align-items: center;
	list-style: none;
	margin: 0;
	padding: 0;
}

/* if there's space, icons are placed alongside the name via js/social-icons.js */
.SocialIcons--top {
	position: absolute;
	left: 0;
	top: 6px;
}
