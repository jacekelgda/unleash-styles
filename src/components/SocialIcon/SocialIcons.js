import React from 'react'
import styles from './SocialIcons.css'

/**
 * SocialIcons
 */
function SocialIcons (props) {
  return (
    <ul className={styles.SocialIcons}>
      {props.children}
    </ul>
  )
}

export default SocialIcons
