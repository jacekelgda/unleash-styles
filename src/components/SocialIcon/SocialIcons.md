SocialIcon component:

    <SocialIcon type='facebook' href='http://google.com' />

SocialIcons component:

    <SocialIcons>
      <SocialIcon type='facebook' href='http://google.com' />
      <SocialIcon type='instagram' href='http://google.com' />
      <SocialIcon type='linkedin' href='http://google.com' />
      <SocialIcon type='pinterest' href='http://google.com' />
      <SocialIcon type='twitter' href='http://google.com' />
    </SocialIcons>
