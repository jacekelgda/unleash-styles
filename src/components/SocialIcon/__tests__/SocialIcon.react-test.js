/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import SocialIcons from '../SocialIcons'
import SocialIcon from '../SocialIcon'

test('SocialIcons renders correctly', () => {
  const wrapper = shallow(
    <SocialIcons>
      <SocialIcon type='facebook' href='http://google.com' />
      <SocialIcon type='instagram' href='http://google.com' />
      <SocialIcon type='linkedin' href='http://google.com' />
      <SocialIcon type='pinterest' href='http://google.com' />
      <SocialIcon type='twitter' href='http://google.com' />
    </SocialIcons>
  )
  expect(wrapper).toMatchSnapshot()
})
