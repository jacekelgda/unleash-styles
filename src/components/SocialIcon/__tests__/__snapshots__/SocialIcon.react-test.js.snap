// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="SocialIcons"
>
    <SocialIcon
        href="http://google.com"
        type="facebook"
    />
    <SocialIcon
        href="http://google.com"
        type="instagram"
    />
    <SocialIcon
        href="http://google.com"
        type="linkedin"
    />
    <SocialIcon
        href="http://google.com"
        type="pinterest"
    />
    <SocialIcon
        href="http://google.com"
        type="twitter"
    />
</ul>,
  "nodes": Array [
    <ul
      className="SocialIcons"
>
      <SocialIcon
            href="http://google.com"
            type="facebook"
      />
      <SocialIcon
            href="http://google.com"
            type="instagram"
      />
      <SocialIcon
            href="http://google.com"
            type="linkedin"
      />
      <SocialIcon
            href="http://google.com"
            type="pinterest"
      />
      <SocialIcon
            href="http://google.com"
            type="twitter"
      />
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SocialIcons>
        <SocialIcon
                href="http://google.com"
                type="facebook"
        />
        <SocialIcon
                href="http://google.com"
                type="instagram"
        />
        <SocialIcon
                href="http://google.com"
                type="linkedin"
        />
        <SocialIcon
                href="http://google.com"
                type="pinterest"
        />
        <SocialIcon
                href="http://google.com"
                type="twitter"
        />
</SocialIcons>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <SocialIcon
              href="http://google.com"
              type="facebook"
/>,
            <SocialIcon
              href="http://google.com"
              type="instagram"
/>,
            <SocialIcon
              href="http://google.com"
              type="linkedin"
/>,
            <SocialIcon
              href="http://google.com"
              type="pinterest"
/>,
            <SocialIcon
              href="http://google.com"
              type="twitter"
/>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="SocialIcons"
>
          <SocialIcon
                    href="http://google.com"
                    type="facebook"
          />
          <SocialIcon
                    href="http://google.com"
                    type="instagram"
          />
          <SocialIcon
                    href="http://google.com"
                    type="linkedin"
          />
          <SocialIcon
                    href="http://google.com"
                    type="pinterest"
          />
          <SocialIcon
                    href="http://google.com"
                    type="twitter"
          />
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="SocialIcons"
>
          <SocialIcon
                    href="http://google.com"
                    type="facebook"
          />
          <SocialIcon
                    href="http://google.com"
                    type="instagram"
          />
          <SocialIcon
                    href="http://google.com"
                    type="linkedin"
          />
          <SocialIcon
                    href="http://google.com"
                    type="pinterest"
          />
          <SocialIcon
                    href="http://google.com"
                    type="twitter"
          />
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SocialIcons>
    <SocialIcon
        href="http://google.com"
        type="facebook"
    />
    <SocialIcon
        href="http://google.com"
        type="instagram"
    />
    <SocialIcon
        href="http://google.com"
        type="linkedin"
    />
    <SocialIcon
        href="http://google.com"
        type="pinterest"
    />
    <SocialIcon
        href="http://google.com"
        type="twitter"
    />
</SocialIcons>,
}
`;
